### PR TITLE
Tests for premisAgentXMLToObject.

### PR DESCRIPTION
**Pull request by damonkelley**
Partially completes #27 

I reworked how the XML is used in the tests. Now that we have more tests that need to share these resources, many of the XML 'fixtures' will be reworked in a similar fashion, and they will likely be relocated to a centrally located module.

### CODE REVIEW

**Comment by damonkelley** [comment]
ping @ldko @somexpert

File: tests/test_presentation.py [hunk 1]
```diff
@@ -197,3 +198,128 @@ def test_datetime_that_has_decimal_seconds(self, event_xml):
...
+        presentation.premisAgentXMLToObject(xml)
+
+        assert get_object_mock.called
+        # If agent_mock was not called, we know that an existing Agent was
+        # retrieved.
+        assert not agent_mock.called
+
+    def test_sets_agent_identifier(self):
+        xml = self.agent_xml(self.identifier)
+        presentation.premisAgentXMLToObject(xml)
```
Comment (somexpert) [inline tests/test_presentation.py]: Duplicate of next statement, minus storing the result in a var. Can be removed.

File: tests/test_presentation.py [hunk 2]
```diff
@@ -197,3 +198,132 @@ def test_datetime_that_has_decimal_seconds(self, event_xml):
...
+
+        assert get_object_mock.called
+        # If agent_mock was not called, we know that an existing Agent was
+        # retrieved.
+        assert not agent_mock.called
+
+    def test_sets_agent_identifier(self, agent_xml):
+        xml = agent_xml(self.identifier)
+        presentation.premisAgentXMLToObject(xml)
```
Comment (ldko) [inline tests/test_presentation.py]: Do we need this line?

**Comment by somexpert** [comment]
Not sure what I think about that last test. I feel like it could be done another way. Gonna think about it a bit.

**Comment by somexpert** [comment]
Eh, the only thing I can think of would be to add an assertion that the returned object was an Agent object, but that doesn't feel quite right. idk, I'm not a fan of tests without assertions, but I can see why that choice was made.

**Comment by damonkelley** [comment]
The assertion is there, it just doesn't use the `assert` keyword.

**Comment by somexpert** [comment]
:thumbsup:

**Comment by ldko** [comment]
:+1:

**Pull request merged**